distributed/scheduler.py

```diff
@@ -5379,16 +5379,27 @@ def adaptive_target(self, comm=None, target_duration=None):
             cpu = max(1, cpu)
 
         # Memory
-        limit_bytes = {addr: ws.memory_limit for addr, ws in self.workers.items()}
+        limit_bytes = [ws.memory_limit for ws in self.workers.values()]
         worker_bytes = [ws.nbytes for ws in self.workers.values()]
-        limit = sum(limit_bytes.values())
+        limit = sum(limit_bytes)
         total = sum(worker_bytes)
-        if total > 0.6 * limit:
+        if total < 0.6 * limit or limit <= 0:
             memory = 2 * len(self.workers)
         else:
             memory = 0
 
         target = max(memory, cpu)
```

Comment:
shouldn't this be `min(memory, cpu)`? If we have not enough CPUs/memory for more workers, we shouldn't spawn more workers

Reply:
I'm not sure I understand.  If we need more memory then we should spawn more workers.  If we need more CPUs then we should spawn more workers.  At least that's my perception of common desire today.

Reply:
sorry, yes you are right.